superset/views.py

```diff
@@ -205,7 +205,8 @@ def check_ownership(obj, raise_if_false=True):
 
 def get_user_roles():
     if g.user.is_anonymous():
-        return [appbuilder.sm.find_role('Public')]
+        public_role = config.get('AUTH_ROLE_PUBLIC', None)
+        return [appbuilder.sm.find_role(public_role)] if public_role is not None else []
```

Comment:
`if public_role else []` should be a bit better, i don't think we want to consider things like an empty string as valid role.